plugins/pulp_rpm/plugins/migrations/0029_checksums_and_templates.py

```diff
@@ -0,0 +1,195 @@
+from cStringIO import StringIO
+import sys
+
+from pulp.server.db import connection
+
+if sys.version < (2, 7):
+    # We need the non-C implementation so we can shove a namespace into its internal data structure,
+    # since in python 2.6, the register_namespace function wasn't present.
+    import xml.etree.ElementTree as ET
+else:
+    import xml.etree.cElementTree as ET
+
+
+RPM_NAMESPACE = 'http://linux.duke.edu/metadata/rpm'
+
+# this is required because some of the pre-migration XML tags use the "rpm"
+# namespace, which causes a parse error if that namespace isn't declared.
+FAKE_XML = '<?xml version="1.0" encoding="%(encoding)s"?><faketag ' \
+           'xmlns:rpm="%(namespace)s">%(xml)s</faketag>'
+
+
+def migrate(*args, **kwargs):
```

Comment:
I had not considered this would come with a migration since `checksums` is not a required field it will hydrate from the DB correctly as `{}` if the Document does not contain the `checksums` attribute.

If we keep the migration it could make upgrades take a long time. If we don't keep it then the first publish of a repo for a checksum type could take a long time.

What about the idea of having this be a separate script the user could run outside of the migration framework?